### PR TITLE
Import history from Safari

### PR DESCRIPTION
**Pull request by simonhong**
Fix https://github.com/brave/brave-browser/issues/8515
Fix https://github.com/brave/brave-browser/issues/8343

image

image

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
See issue.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: browser/ui/webui/settings/brave_import_data_handler_mac.mm [hunk 1]
```diff
@@ -32,6 +32,26 @@
 
 using content::BrowserThread;
 
+class ScopedOpenFile {
+ public:
+  explicit ScopedOpenFile(const base::FilePath& file_path) {
+    file_ = base::OpenFile(file_path, "r");
```
Comment (simonhong) [inline browser/ui/webui/settings/brave_import_data_handler_mac.mm:38]: Just trying to open files that need full disk access adds Brave browser to full disk access panel list.

**Comment by simonhong** [comment]
Kindly ping to reviewers :)

File: utility/importer/brave_safari_importer.mm [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+#include "base/strings/utf_string_conversions.h"
+#include "base/time/time.h"
+#include "chrome/common/importer/importer_bridge.h"
+#include "chrome/common/importer/importer_url_row.h"
+#include "sql/statement.h"
+#include "url/gurl.h"
+
+BraveSafariImporter::~BraveSafariImporter() = default;
+
+void BraveSafariImporter::ImportHistory() {
```
Comment (bridiver) [inline utility/importer/brave_safari_importer.mm:24]: I'm confused, why are we doing this if chrome already imports the history?

Comment (simonhong) [reply utility/importer/brave_safari_importer.mm:24]: Chrome's history importing from Safari is broken. filed issue to https://bugs.chromium.org/p/chromium/issues/detail?id=1078626
The reason is Chrome still check `History.plist` but that file is not existed anymore and `History.db` is introduced.

Comment (bsclifton) [reply utility/importer/brave_safari_importer.mm:24]: Nice! This is one we can upstream maybe?

Comment (simonhong) [reply utility/importer/brave_safari_importer.mm:24]: I'll try :)

Comment (simonhong) [reply utility/importer/brave_safari_importer.mm:24]: @bridiver Do you have more concerns for this PR?

**Pull request merged**